### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by OSUANSDNC** [importZL/BLO-SAM#1]
This is a great work, may I ask if the model is learning both datasets synchronously while training? It seems to me from the code that only one dataset is trained, but in the framework diagram it is two datasets trained synchronously?

**Comment by importZL** [comment]
Actually, there are two groups of learnable parameters: the parameter of LoRA layers, and the prompt embedding. The first one is learned on the training set, and the second one is learned on the validation set.

**Comment by importZL** [comment]
Actually, there are two groups of learnable parameters: the parameter of LoRA layers, and the prompt embedding. The first one is learned on the training set, and the second one is learned on the validation set.
